app/client/cli/actor.go

```diff
@@ -0,0 +1,318 @@
+package cli
+
+import (
+	"fmt"
+	"math/big"
+	"regexp"
+	"strings"
+
+	"github.com/pokt-network/pocket/shared/converters"
+	typesUtil "github.com/pokt-network/pocket/utility/types"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(NewActorCommands(attachPwdFlagToSubcommands())...)
+
+	rawChainCleanupRegex = regexp.MustCompile(rawChainCleanupExpr)
+
+	oneMillion = big.NewInt(1e6)
+}
+
+const (
+	// DISCUSS(team): this should probably come from somewhere else
+	stakingRecommendationAmount = 15100000000
+	rawChainCleanupExpr         = "[^,a-fA-F0-9]+"
+)
+
+var (
+	pwd                  string
+	rawChainCleanupRegex *regexp.Regexp
+	oneMillion           *big.Int
+)
+
+type (
+	cmdOption   func(*cobra.Command)
+	actorCmdDef struct {
+		Name      string
+		ActorType typesUtil.ActorType
+		Options   []cmdOption
+	}
+)
+
+func NewActorCommands(cmdOptions []cmdOption) []*cobra.Command {
+	actorCmdDefs := []actorCmdDef{
+		{"Application", typesUtil.ActorType_App, cmdOptions},
+		{"Node", typesUtil.ActorType_ServiceNode, cmdOptions},
+		{"Fisherman", typesUtil.ActorType_Fisherman, cmdOptions},
+		{"Validator", typesUtil.ActorType_Validator, cmdOptions},
+	}
+
+	cmds := make([]*cobra.Command, len(actorCmdDefs))
+	for i, cmdDef := range actorCmdDefs {
+		cmd := &cobra.Command{
+			Use:     cmdDef.Name,
+			Short:   fmt.Sprintf("%s actor specific commands", cmdDef.Name),
+			Aliases: []string{strings.ToLower(cmdDef.Name), cmdDef.ActorType.GetName()},
+			Args:    cobra.ExactArgs(0),
+		}
+		cmd.AddCommand(newActorCommands(cmdDef)...)
+		cmds[i] = cmd
+	}
+	return cmds
+}
+
+func newActorCommands(cmdDef actorCmdDef) []*cobra.Command {
+	cmds := []*cobra.Command{
+		newStakeCmd(cmdDef),
+		newEditStakeCmd(cmdDef),
+		newUnstakeCmd(cmdDef),
+		newUnpauseCmd(cmdDef),
+	}
+	applySubcommandOptions(cmds, cmdDef)
+	return cmds
+}
+
+func newStakeCmd(cmdDef actorCmdDef) *cobra.Command {
+	stakeCmd := &cobra.Command{
+		Use:   "Stake",
+		Short: fmt.Sprintf("Stake an actor (%s) in the network.", cmdDef.Name),
+		Long:  fmt.Sprintf("Stake the %s actor into the network, making it available for service.", cmdDef.Name),
+	}
+
+	custodialStakeCmd := &cobra.Command{
```

Comment:
custodial stake command is not an option in V1 - please remove

Reply:
Also corresponding doc ^

Reply:
In that case, shall we remove the stake command altogether then? If we remove custodial, stake won't have any sub-commands.

Or maybe, should we implement non-custodial?

Reply:
```
message MessageStake {
  ActorType actor_type = 1;
  bytes public_key = 2;
  repeated string chains = 3;
  string amount = 4;
  string service_url = 5;
  bytes output_address = 6;
  optional bytes signer = 7;
}
```

Here is the stake message. All stake commands are "implicitly" custodial if `output_adress` is an empty string, so no need for the explicit subcommand.

Reply:
Addressed in https://github.com/pokt-network/pocket/pull/177/commits/c9efa96cd17e74f505847b4287c3465e8c8b9eeb